gateway/src/project.rs

```diff
@@ -1402,7 +1430,25 @@ where
             }));
         }
 
-        let new_stat = get_container_stats(&container).await?;
+        let new_stat = match ctx.stats_source() {
+            DockerStatsSource::CgroupV1 => get_container_stats_cgroup_v1(&container).await?,
+            DockerStatsSource::CgroupV2 => get_container_stats_cgroup_v2(&container).await?,
+            DockerStatsSource::Bollard => {
+                let new_stat = ctx
+                    .docker()
+                    .stats(
+                        safe_unwrap!(container.id),
+                        Some(StatsOptions {
+                            one_shot: true,
+                            stream: false,
+                        }),
+                    )
+                    .next()
+                    .await
+                    .unwrap()?;
+                new_stat.cpu_stats.cpu_usage.total_usage
+            }
```

Comment:
What do you think of moving that to the `impl` of `DockerStatsSource`, or even of `DockerContext` itself (well, inside their implementation) ? And instead just have `ctx.get_stats(&container)` here. 

In addition to the encapsulation of the stats feature, it would help to have a single common span name for the instrumentation + an additional field that represents its version. Currently `get_container_stats_cgroup_v2` and `get_container_stats_cgroup_v1` have a different name, and if we call the `bollard` version, there is no span AFAIK. If you're going the route of having a single `get_stats` function, you'll just have one place to instrument.